packages/cms-data-sync/src/utils/migration.ts

```diff
@@ -57,9 +43,39 @@ export const migrateFromSquidexToContentfulFactory =
           n += 1;
           logger(`Created entry with id ${id}. (${n}/${data.length})`, 'INFO');
           if (item.status === 'PUBLISHED') {
-            return entry;
+            return createdEntry;
           }
+          return null;
+        };
+
+        const updateExistingEntry = async () => {
+          const contentfulEntry = await contentfulEnvironment.getEntry(id);
+          updateEntryFields(contentfulEntry, payload);
+          const updatedEntry = await contentfulEntry.update();
+          await contentfulRateLimiter.removeTokens(1);
+
+          n += 1;
+          logger(`Updated entry with id ${id}. (${n}/${data.length})`, 'INFO');
+          return updatedEntry;
+        };
+
+        const shouldUpdateEntry = updateEntry || isUpdateModeEnabled;
+
+        try {
+          return shouldUpdateEntry
+            ? await updateExistingEntry()
+            : await createEntry();
         } catch (err) {
+          if (err instanceof Error) {
+            const errorParsed = JSON.parse(err.message);
+            // this is a fallback when it should have updated the entry
+            // but it does not exist
+            if (shouldUpdateEntry && errorParsed.status === 404) {
```

Comment:
I'm not 100% sure that we want to fall back to creation if the `parseData` function returns an `updateEntry` property. The previous implementation did not. 

```suggestion
            if (isUpdateModeEnabled && errorParsed.status === 404) {
```

Reply:
good catch, I'll change that to your suggestion!